Lib/pickle.py

```diff
@@ -336,22 +338,21 @@ def whichmodule(obj, name):
             except AttributeError:
                 pass
         module_name = '__main__'
-    elif module_name is None:
-        module_name = '__main__'
 
     try:
         __import__(module_name, level=0)
         module = sys.modules[module_name]
+    except (ImportError, ValueError, KeyError) as exc:
+        raise PicklingError(f"Can't pickle {obj!r}: {exc!s}")
```

Comment:
Should we do `from None` or `from exc` here (and in the other cases below?)

Reply:
No, removing `from None` was intentional. Error here is not normal. The original exception and its traceback can add additional context.

We use `from None` if the details of the original exception are not important. For example when it is a result of using EAFP instead of LBYL. But in this case we expect that the object can be found by name, and if this fails, we want to know gore details.